File: app/src/main/java/com/geridea/trentastico/gui/views/requestloader/ExtraCoursesLoadingMessage.java
package com.geridea.trentastico.gui.views.requestloader;

import com.geridea.trentastico.model.ExtraCourse;
import com.geridea.trentastico.network.controllers.LessonsController;
import com.geridea.trentastico.utils.time.WeekInterval;

public class ExtraCoursesLoadingMessage extends AbstractTextMessage {

    private final WeekInterval interval;

    private final ExtraCourse course;
    private final boolean isARetry;

    public ExtraCoursesLoadingMessage(LessonsController.ExtraLessonsRequest request) {
        super(request.getOperationId());

        interval = request.getIntervalToLoad();

        this.course = request.getExtraCourse();
        this.isARetry = request.isRetrying();
    }


    @Override
    public String getText() {
        String format = isARetry ?
                 "Sto riprovando a scaricare gli orari del corso \"%s\" dal %s..."
               : "Sto scaricando gli orari del corso \"%s\" dal %s...";

        return String.format(format, course.getName(), formatFromToString(interval));
    }

}

File: app/src/main/java/com/geridea/trentastico/gui/views/requestloader/ExtraCoursesLoadingMessage.kt
package com.geridea.trentastico.gui.views.requestloader

import com.geridea.trentastico.model.ExtraCourse
import com.geridea.trentastico.network.controllers.LessonsController
import com.geridea.trentastico.utils.time.WeekInterval

class ExtraCoursesLoadingMessage(request: LessonsController.ExtraLessonsRequest) : AbstractTextMessage(request.operationId) {

    private val interval: WeekInterval = request.intervalToLoad
    private val course: ExtraCourse    = request.extraCourse
    private val isARetry: Boolean      = request.isRetrying


    override val text: String
        get() {
            val format = if (isARetry)
                 "Sto riprovando a scaricare gli orari del corso \"%s\" dal %s..."
            else
                "Sto scaricando gli orari del corso \"%s\" dal %s..."

            return String.format(format, course.name, formatFromToString(interval))
        }

}
